src/processFile.ts

```diff
@@ -75,21 +75,17 @@ export function processSourceFile(src: SourceFile, options: ProcessFileOptions):
 function adjustImportExportStatements(src: SourceFile, options: ProcessFileOptions, magicString: MagicString) {
     const { srcFilename } = src;
     const { root: srcRoot, target: targetRoot, allowJsOutsideOfRoot, ext } = options;
-    const content = magicString.toString();
-
-    const exp = new RegExp(regExpImportExport);
 
     let linesChanged = 0;
-    let match: RegExpExecArray | null;
-    while ((match = exp.exec(content))) {
-        const { index } = match;
+    const content = magicString.toString();
 
-        const line = match[0];
-        const reference = match.groups?.['file'];
+    // Use replace with a callback to handle all matches at once
+    const processedContent = content.replace(new RegExp(regExpImportExport, 'g'), (match, ...args) => {
+        // Extract groups from the match - the groups are at the end of args array
+        const groups = args[args.length - 1];
+        const reference = groups?.['file'];
```

Comment:
```suggestion
        const reference = groups?.['file'];
        const pos = match.indices?.groups?.['file'];
```

Extract the position of the match.